Storage/SQL/SQLiteLogins.swift

```diff
@@ -535,43 +535,45 @@ public class SQLiteLogins: BrowserLogins {
         if guids.isEmpty {
             return succeed()
         }
-
+        
+        var statements: [(sql: String, args: Args?)] = []
         let nowMillis = NSDate.now()
-        let inClause = BrowserDB.varlist(guids.count)
-
-        // Immediately delete anything that's marked as new -- i.e., it's never reached
-        // the server.
-        let delete =
-        "DELETE FROM \(TableLoginsLocal) WHERE guid IN \(inClause) AND sync_status = \(SyncStatus.New.rawValue)"
-
-        // Otherwise, mark it as changed.
-        let update =
-        "UPDATE \(TableLoginsLocal) SET " +
-        " local_modified = \(nowMillis)" +
-        ", sync_status = \(SyncStatus.Changed.rawValue)" +
-        ", is_deleted = 1" +
-        ", password = ''" +
-        ", hostname = ''" +
-        ", username = ''" +
-        " WHERE guid IN \(inClause)"
-
-       let markMirrorAsOverridden =
-        "UPDATE \(TableLoginsMirror) SET " +
-            "is_overridden = 1 " +
-            "WHERE guid IN \(inClause)"
-
-        let insert =
-        "INSERT OR IGNORE INTO \(TableLoginsLocal) " +
-            "(guid, local_modified, is_deleted, sync_status, hostname, timeCreated, timePasswordChanged, password, username) " +
-        "SELECT guid, \(nowMillis), 1, \(SyncStatus.Changed.rawValue), '', timeCreated, \(nowMillis)000, '', '' FROM \(TableLoginsMirror) WHERE guid IN \(inClause)"
-
-        let args: Args = guids.map { $0 as AnyObject }
-        return self.db.run([
```

Comment:
One way you can make this change shorter and easier to understand is to pull out the whole chunk you just reindented as a function:

```
private func getDeletionStatementsForGUIDs(guids: [GUID], timestamp: Timestamp) {
    let inClause = …
    …
    return [
        (delete, args),
       …
    ]
}
```

Then this function becomes incredibly short; something like this:

```
let now = NSDate.now()
this.db.run(chunk(guids, by: BrowserDB.MaxVariableNumber).flatMap({ this.getDeletionStatementsForGUIDs($0, now) }))…
```